cpp/src/io/orc/writer_impl.cu

```diff
@@ -2454,16 +2455,17 @@ void writer::impl::write(table_view const& input)
   // is still intact.
   // Note that `out_sink_` is intentionally passed by const reference to prevent accidentally
   // writing anything to it.
-  auto [streams,
-        comp_results,
-        strm_descs,
-        enc_data,
-        segmentation,
-        stripes,
-        orc_table,
-        compressed_data,
-        intermediate_stats,
-        stream_output] = [&] {
+  [[maybe_unused]] auto [streams,
+                         comp_results,
+                         strm_descs,
+                         enc_data,
+                         segmentation,
+                         stripes,
+                         orc_table,
+                         pd_masks, /* unused, but needs to be kept alive */
```

Comment:
Can you help me understand why the lifetime matters if it's unused? I don't see what this is holding, or what is holding this data, that makes its lifetime relevant.

Reply:
Why is it needed after `convert_table_to_orc_data`? statistics computation?

Reply:
It is not used, but it stores null mask for `orc_table` to be valid. So as long as we still need `orc_table` we need to keep it.

Reply:
> So as long as we still need `orc_table` we need to keep it.

Not exactly. We only need it as long as we need the null masks in the `orc_table`. I can't find where that is used `convert_table_to_orc_data`.

Reply:
It is used in `calculate_aligned_rowgroup_bounds` which is called in `encode_columns`.